src/services/outliningElementsCollector.ts

```diff
@@ -2,13 +2,17 @@
 namespace ts.OutliningElementsCollector {
     const collapseText = "...";
     const maxDepth = 20;
+    const defaultLabel = "#region";
+    const regionMatch = new RegExp("^\\s*//\\s*(#region|#endregion)(?:\\s+(.*))?$");

```

Comment:
consider making `end` optional, this way you can just check on the truthiness of  `result[1]`
`new RegExp("^\\s*//\\s*#(end)?region(?:\\s+(.*))?$");`